File: 05_FitTrack/Javascript/fitTrack.js
let Z_Accel = 0
let Y_Accel = 0
let X_Accel = 0
let measuring = false
radio.onDataPacketReceived( ({ receivedString: name, receivedNumber: value }) =>  {
    if (name == "x") {
        X_Accel = value
    } else if (name == "y") {
        Y_Accel = value
    } else if (name == "z") {
        Z_Accel = value
        write_data()
    } else if (name == "done") {
        serial.writeLine("done")
        basic.showIcon(IconNames.Happy)
    }
})
input.onButtonPressed(Button.AB, () => {
    measuring = false
    radio.sendValue("done", 0)
    basic.showIcon(IconNames.Happy)
})
input.onButtonPressed(Button.A, () => {
    measuring = true
    basic.showIcon(IconNames.Heart)
})
input.onButtonPressed(Button.B, () => {
    measuring = false
    basic.showIcon(IconNames.Yes)
})
function capture_data()  {
    X_Accel = input.acceleration(Dimension.X)
    Y_Accel = input.acceleration(Dimension.Y)
    Z_Accel = input.acceleration(Dimension.Z)
    radio.sendValue("x", X_Accel)
    radio.sendValue("y", Y_Accel)
    radio.sendValue("z", Z_Accel)
}
function write_data()  {
    serial.writeLine("" + X_Accel + "," + Y_Accel + "," + Z_Accel)
}
radio.setGroup(1)
radio.setTransmitPower(7)
basic.showIcon(IconNames.Yes)
basic.forever(() => {
    if (measuring) {
        capture_data()
        basic.pause(200)
    }
})

File: 05_FitTrack/Python/fitTrack.py
from microbit import *
import radio

radio.on()
radio.config(group=1)
radio.config(power=7)
z_Accel = 0
y_Accel = 0
x_Accel = 0
measuring = False
display.show(Image.YES)

def capture_data():
    vals = {'x':0, 'y':0, 'z':0}
    vals['x'] = accelerometer.get_x()
    vals['y'] = accelerometer.get_y()
    vals['z'] = accelerometer.get_z()
    radio.send("x"+str(x_Accel))
    radio.send("y"+str(y_Accel))
    radio.send("z"+str(z_Accel))
    
        
def write_data():
    print(x_Accel, y_Accel, z_Accel, sep=',', end='\n')

while True:
    if button_a.is_pressed() and button_b.is_pressed():
        measuring = False
        radio.send("done")
        display.show(Image.HAPPY)
    elif button_a.is_pressed():
        measuring = True
        display.show(Image.HEART)
    elif button_b.is_pressed():
        measuring = False
        display.show(Image.YES)
    incoming = radio.receive()
    if incoming is not None:
        if incoming.startswith('done'):
            display.show(Image.HAPPY)
            print("done")
        elif incoming.startswith('x'):
        elif incoming.startswith('y'):
        elif incoming.startswith('z'):
            
            write_data()
        else:
            #x_Accel, y_Accel, z_Accel = incoming.split(",")
            #write_data()
    if measuring:
        capture_data()
    sleep(200)